lib/carrierwave/storage/fog.rb

```diff
@@ -383,12 +385,19 @@ def public_url
                 use_virtual_hosted_style = @uploader.fog_directory.to_s =~ subdomain_regex && !(protocol == 'https' && @uploader.fog_directory =~ /\./)
 
                 region = @uploader.fog_credentials[:region].to_s
-                regional_host = case region
-                                when DEFAULT_S3_REGION, ''
-                                  's3.amazonaws.com'
-                                else
-                                  "s3.#{region}.amazonaws.com"
-                                end
+                regional_host = 's3.amazonaws.com' # used for DEFAULT_S3_REGION or no region set
+                if ENV['AWS_USE_FIPS_ENDPOINT'] == 'true' && AWS_FIPS_REGIONS.include?(region)
```

Comment:
Don't use an environment variable here. You should create a config like `fog_aws_fips`, in a same manner with `fog_aws_accelerate`.  You don't need to check AWS region here, just letting developer choose is enough.

Reply:
This is the same environment variable that the AWS SDK uses to determine if _it_ should choose the FIPS endpoints, hence my use of it here.

Reply:
CarrierWave is not the AWS SDK. Every software has its own way of configuration, and you need to follow it. Otherwise there'll be a mess.